src/main/java/com/chriscarini/jetbrains/gitpushreminder/GitPushReminder.java

```diff
@@ -1,53 +1,85 @@
+            return false;
+        }
+
+        return true;
+    }
+
+    private boolean doNotCloseOnUnpushedCommits(@NotNull Project project) {
         final List<GitHelper.RepositoryAndBranch> branchesWithUnpushedCommits = GitHelper.getBranchesWithUnpushedCommits(project, SettingsManager.getInstance().getState().checkAllBranches);
 
-        // If there are *NO* branches with outgoing/un-pushed commits, then we can close.
-        boolean canClose = branchesWithUnpushedCommits.isEmpty();
+        if (branchesWithUnpushedCommits.isEmpty()) {
+            // If there are *NO* branches with outgoing/un-pushed commits, then we can close.
+            return false;
+        }
 
-        if (!canClose && SettingsManager.getInstance().getState().showDialog) {
-            final int dialogResult = Messages.showOkCancelDialog(project,
+        final int dialogResult = Messages.showOkCancelDialog(project,
                 PluginMessages.get("git.push.reminder.closing.dialog.body.unpushed.branches",
-                    branchesWithUnpushedCommits.stream()
-                            .sorted((o1, o2) -> {
-                                if (o1.repository().getRoot().getName().equals(o2.repository().getRoot().getName())) {
-                                    return o1.branch().getName().compareTo(o2.branch().getName());
-                                }
-                                return o1.repository().getRoot().getName().compareTo(o2.repository().getRoot().getName());
-                            })
-                        .map(repoAndBranch -> String.format(
-                                "%s (%s)", //NON-NLS
-                                repoAndBranch.branch().getName(),
-                                repoAndBranch.repository().getRoot().getName()
-                            )
-                         )
-                        .collect(Collectors.joining("</li><li>"))
+                        branchesWithUnpushedCommits.stream()
+                                .sorted((o1, o2) -> {
+                                    if (o1.repository().getRoot().getName().equals(o2.repository().getRoot().getName())) {
+                                        return o1.branch().getName().compareTo(o2.branch().getName());
+                                    }
+                                    return o1.repository().getRoot().getName().compareTo(o2.repository().getRoot().getName());
+                                })
+                                .map(repoAndBranch -> String.format(
+                                                "%s (%s)", //NON-NLS
+                                                repoAndBranch.branch().getName(),
+                                                repoAndBranch.repository().getRoot().getName()
+                                        )
+                                )
+                                .collect(Collectors.joining("</li><li>"))
```

Comment:
let's keep these lines untouched, since the delta is only additional whitespace

Reply:
The `&& SettingsManager.getInstance().getState().showDialog` was extracted, and now is done before calling the method. If I would revert the other changes in this block, it would result in:
```
        boolean canClose = branchesWithUnpushedCommits.isEmpty();

        if (!canClose) {
```
which is very ugly as there is only one usage of the `canClose` var.